### ISSUE TITLE
22.0.0: crl.get_revoked() fails with AttributeError

### ISSUE THREAD

**Issue by grindsa** [pyca/pyopenssl#1091]
Hi,

Since beginning of February I see in my weekly regression the below error .

```
[wsgi:error] [pid 15:tid 139849151395584] [remote 172.18.0.3:38622]     crl_list = crl.get_revoked()
[wsgi:error] [pid 15:tid 139849151395584] [remote 172.18.0.3:38622]   File "/usr/local/lib/python3.8/dist-packages/OpenSSL/crypto.py", line 2315, in get_revoked
[wsgi:error] [pid 15:tid 139849151395584] [remote 172.18.0.3:38622]     revoked_copy = _lib.X509_REVOKED_dup(revoked)
[wsgi:error] [pid 15:tid 139849151395584] [remote 172.18.0.3:38622] AttributeError: module 'lib' has no attribute 'X509_REVOKED_dup'
```
I think there the problem has been introduced with pyopenssl 22.0.0. Any idea what is going wrong or any hint how to troubleshoot this problem further?

Thank you and best regards,
/GrindSa

**Comment by alex** [comment]
What version of `cryptography` do you have installed?

**Comment by grindsa** [comment]
Hi,

I am using the packages shipped with Ubuntu 20.04 which is `python3-cryptography amd64 2.8-3ubuntu0.1`

But thanks for pointing me to this. I just found the below error in the logs generated during container creation.

```
ERROR: pyopenssl 22.0.0 has requirement cryptography>=35.0, but you'll have cryptography 2.8 which is incompatible.
```

I will update cryptography and give it another try.

Thx

**Comment by alex** [comment]
You should not mix packages from ubuntu and from pip. I recommend you install all your packages using pip into a virtualenv.
